v2/client.go

```diff
@@ -41,6 +42,9 @@ func NewClient(config *ClientConfiguration) (Client, error) {
 	if config.Insecure {
```

Comment:
OK. I have added checks to make ensure that TLSClientConfig.InsecureSkipVerify and TLSClientConfig.RootCAs are not in conflict with each other.

Or would it be better to clear out TLSClientConfig.RootCAs with TLSClientConfig.InsecureSkipVerify is true? This would effectively ignore any root CAs that the user supplies, with the assumption that the user wants to ignore all TLS verification anyway.